### DISCUSSION TITLE
Why Facebook Robyn Algorithm needs only unique dates?

### DISCUSSION THREAD

**Discussion by Anil-Pothala** [facebookexperimental/Robyn#417] [General]
My team and I have been exploring the Robyn algorithm to do Market Mix Modeling on my dataset. The dataset is monthly level data of the promotional activity for each customer. In this case, the data looks like this-

|Id                   |Date               |Revenue  |Channels…|
|--------------  |---------------|-----------|------------|
|Customer 1    |Jan-2021       |                |                 |        
|   …..               |                      |                |                 |  
|Customer 1    |Dec-2021      |                |                 |
|Customer 2    |Feb-2021      |                 |                 |
|   …..               |                      |                 |                 |
|Customer 2     |Dec-2021      |                |                 |

In this way, we have over 1000 customers and their monthly data of the channel activity. We have been able to create models using linear regression to get the impact of each channel. Now when we tried to run this data on Robyn we get a duplicate date error, so does this mean we have to run Robyn algorithm for each customer separately? Then we will have only 12 data points for the model and getting daily or weekly is also not possible for us. Is there any way to run this kind of data on Robyn? Also, why does Robyn restrict us to use unique dates even though it uses ridge regression internally which would not be affected by dates, isn’t having more data points better?

**Comment by laresbernardo** [comment]
[Answered](https://github.com/facebookexperimental/Robyn/issues/418).
